aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1014,6 +1014,14 @@ Tensor& logsumexp_out(const Tensor& self, DimnameList dims, bool keepdim, Tensor
   return at::logsumexp_out(result, self, dimnames_to_positions(self, dims), keepdim);
 }
 
+// special_logsumexp, alias for logsumexp
+Tensor special_logsumexp(const Tensor& self, IntArrayRef dims, bool keepdim) {
```

Comment:
Good point! @mruberry had pointed the similar thing, initially. IIRC, there was an issue with supporting inplace arguments for operators unary operators like `erf`, etc. But for operators like `logsumexp` this does make sense and should be straightforward to do so. 

It would be nice to make `special_logsumexp` the primary implementation. (We can defer to doing it in a later PR as well)

Thanks for raising this point @Lezcano 

Details:
`special` variants support function and out variants. So to still support  inplace variant for lets say `erf`, we will have to use `special_erf_out` but that won't have a derivative formula so we will have to explicitly put an entry for `erf_` or use erf with copy for inplace variant.
Neither of them seemed convincing to make those changes. So we stuck to defer till the need arises for unary operators.

Reply:
Sorry, but I did not understand. Why does `special_erf_out` not have a backwards formula?

Reply:
`out` variants don't get the backwards formula.

Reply:
Thanks for the discussion, @kshitij12345 and @Lezcano. Since this PR also blocks `polygamma` PR (because of changes in `alias_remapping` test), I'll like to make the `special_logsumexp` implementation as the primary one in another PR.